### PR TITLE
[KernelGen][Nvidia] Add _resize_output_ operator

### PR DESCRIPTION
**Pull request by NineAnnAnn**
## Local validation summary (NVIDIA H20)

I ran the PR checks locally on the current PR head `4d31e2fa5a0cfba3e6c232703cf3f62d8f93fce9` without modifying the PR code.

### Comment status
- GitHub review comments: 0
- GitHub PR-level comments: 0
- GitHub reviews: 0

### Commands run
- `python /root/baai-internship/skills/flaggems-pr-submit/scripts/check_operator.py _resize_output_ --repo-dir /root/FlagGems --strict`
- `CUDA_VISIBLE_DEVICES=0 pytest tests/test__resize_output_.py -q`
- `CUDA_VISIBLE_DEVICES=0 pytest benchmark/test__resize_output_.py --level core -s`

### Results
- Accuracy: PASS (`15 passed in 0.64s`)
- Benchmark: PASS (`1 passed in 29.23s`)
- Strict operator check: FAIL (`5 errors`, `2 warnings`)

### Strict checker failures to address
- Kernel file is missing the required copyright/license header or KernelGen comment.
- `operators.yaml` is missing `id: _resize_output_`.
- The test file is missing `@pytest.mark.underscore_resize_output_`.
- The benchmark file is missing `@pytest.mark.underscore_resize_output_`.
- Branch merge check failed against `upstream/master` with `fatal: refusing to merge unrelated histories`.

### Strict checker warnings
- `ops/__init__.py` import order may need isort.
- In-place test `test_resize_output_` only compares the returned value; the checker suggests also comparing the mutated input.

#### resize_output_ (in-place)

| dtype | Size | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|---|---|---:|---:|---:|
| float16 | `([torch.Size([1073741824]), [1073741824]], {'device': 'cuda'})` | 2.352480 | 2.489824 | 0.945x |
| float16 | `([torch.Size([64, 64]), [64, 64]], {'device': 'cuda'})` | 0.033568 | 0.102016 | 0.329x |
| float16 | `([torch.Size([4096, 4096]), [16777216]], {'device': 'cuda'})` | 0.047328 | 0.096224 | 0.492x |
| float16 | `([torch.Size([64, 512, 512]), [16777216]], {'device': 'cuda'})` | 0.047456 | 0.094784 | 0.501x |
| float16 | `([torch.Size([1024, 1024, 1024]), [1073741824]], {'device': 'cuda'})` | 2.406048 | 2.488384 | 0.967x |
| float32 | `([torch.Size([1073741824]), [1073741824]], {'device': 'cuda'})` | 4.423856 | 4.276848 | 1.034x |
| float32 | `([torch.Size([64, 64]), [64, 64]], {'device': 'cuda'})` | 0.034848 | 0.100352 | 0.347x |
| float32 | `([torch.Size([4096, 4096]), [16777216]], {'device': 'cuda'})` | 0.077680 | 0.105888 | 0.734x |
| float32 | `([torch.Size([64, 512, 512]), [16777216]], {'device': 'cuda'})` | 0.077792 | 0.104480 | 0.745x |
| float32 | `([torch.Size([1024, 1024, 1024]), [1073741824]], {'device': 'cuda'})` | 4.390288 | 4.275392 | 1.027x |
| bfloat16 | `([torch.Size([1073741824]), [1073741824]], {'device': 'cuda'})` | 2.371776 | 2.479840 | 0.956x |
| bfloat16 | `([torch.Size([64, 64]), [64, 64]], {'device': 'cuda'})` | 0.033568 | 0.099200 | 0.338x |
| bfloat16 | `([torch.Size([4096, 4096]), [16777216]], {'device': 'cuda'})` | 0.047264 | 0.096160 | 0.492x |
| bfloat16 | `([torch.Size([64, 512, 512]), [16777216]], {'device': 'cuda'})` | 0.047360 | 0.095104 | 0.498x |
| bfloat16 | `([torch.Size([1024, 1024, 1024]), [1073741824]], {'device': 'cuda'})` | 2.402656 | 2.471040 | 0.972x |

Arithmetic mean speedup: **0.692x**

### Multi-backend Testing
| Backend | Accuracy Test | Speedup (mean) | Notes |
|---|---|---:|---|
| Nvidia (H20) | PASS (15 cases) | 0.692x | Primary local run |
| Tianshu | N/A | — | Not run locally |
| Muxi | N/A | — | Not run locally |
| Ascend | N/A | — | Not run locally |
| Hygon | N/A | — | Not run locally |

### Files changed in this PR
- `src/flag_gems/ops/_resize_output_.py`: Triton kernel implementation
- `tests/test__resize_output_.py`: Accuracy test
- `benchmark/test__resize_output_.py`: Performance benchmark
- `src/flag_gems/ops/__init__.py`: Register import and `__all__`
- `src/flag_gems/__init__.py`: Register to `_FULL_CONFIG`
- `conf/operators.yaml`: Add operator entry

### CODE REVIEW

File: src/flag_gems/ops/_resize_output_.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import logging
+from typing import List
+
+import torch
+
+from flag_gems.ops._resize_output import _resize_output
+
+logger = logging.getLogger(__name__)
+
+
```
Comment (Caeruleann) [inline src/flag_gems/ops/_resize_output_.py:24]: So it's not a triton operator?

Comment (NineAnnAnn) [reply src/flag_gems/ops/_resize_output_.py:24]: The non-in-place version _resize_output.py already merged in master follows the same pattern — it's a PyTorch wrapper that delegates data copying to flag_gems.ops.copy (which uses Triton internally). resize is fundamentally a memory reallocation operation, not a compute operation, so there's no element-wise computation to accelerate with a Triton kernel. My in-place version is consistent with the existing implementation.

**Review by Caeruleann — APPROVED** [review]
